### ISSUE TITLE
Any progress on export yolov5 int tflite with >= tensorflow 2.13?

### ISSUE THREAD

**Issue by ozkulah** [ultralytics/yolov5#13659]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I work with yolov5n and realized warning regarding the export int8 tflite. [warning](https://github.com/ultralytics/yolov5/issues/12489).
Whole system is using tf 2.19 and export just creates a corrupted tflite model. We tried to use an older docker image from repo but couldnt find a proper one for that operation.

Is there any plan to fix that bug in future or can you adress me a possible good docker image which will not have issue related to export int tflite?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ozkulah, thank you for your interest in YOLOv5 🚀! This is an automated response to help you get unstuck quickly — an Ultralytics engineer will also assist you here soon.

Please visit our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/) to get started, where you can find quickstart guides for simple tasks like [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) all the way to advanced concepts like [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/).

If this is a 🐛 Bug Report (e.g., corrupted int8 TFLite exports), please provide a minimum reproducible example to help us debug it:
-1. Exact export command used (e.g., `python export.py --weights yolov5n.pt --include tflite --int8`)
-2. Full console logs and warnings
-3. YOLOv5 commit hash, Python, PyTorch, TensorFlow/TFLite versions, OS, and CUDA info
-4. Small sample/calibration images if using int8, plus details on how the model is “corrupted” (load errors, runtime errors, incorrect outputs, etc.)
-5. If using Docker, the exact image tag or digest

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

## Requirements
[**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Environments
YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>
If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

🙏 Once you share the MRE and environment details, we’ll help investigate next steps for int8 TFLite exports with your setup.

**Comment by ozkulah** [comment]
this a question yes, regarding a known issue in code. Asking for update for that

**Comment by pderrenger** [comment]
Hi @ozkulah — INT8 TFLite export in YOLOv5 is currently unreliable on TensorFlow ≥2.13; the known-good workaround is to pin TensorFlow to 2.12.x for export. Example in Docker: docker run -it --ipc=host ultralytics/yolov5:latest bash -lc "pip install 'tensorflow==2.12.*' tflite-support && python export.py --weights yolov5n.pt --include tflite --int8 --data data/coco128.yaml" — please update to the latest YOLOv5 repo and try this flow. See the YOLOv5 export guide and Docker quickstart for details; we’re tracking TF-side changes but don’t have an ETA yet, and PRs are welcome. See the YOLOv5 export docs and Docker quickstart for details. ([github.com](https://github.com/ultralytics/yolov5/issues/12609), [docs.ultralytics.com](https://docs.ultralytics.com/yolov5/tutorials/model_export/))

**Comment by ozkulah** [comment]
thanks for the info. 
Tensorflow <= 2.13 requires python <3.12 too so , I am not sure using latest and installing tensorflow==2.12.* will work. 
Other lib version dependecies also changing with python .

**Comment by pderrenger** [comment]
You’re right—TF ≤2.13 needs Python <3.12; the simplest fix is to run export in a small, pinned “exporter” env (Py 3.10/3.11 + TF 2.12 + numpy<2), then use the generated .tflite in your main project; for example from the YOLOv5 repo root: 
```
docker run --rm -it -v $PWD:/ws -w /ws python:3.11-slim bash -lc \
"apt-get update && apt-get install -y git && pip install -U pip && \
 pip install 'tensorflow==2.12.*' 'numpy<2' tflite-support && \
 pip install 'torch>=2.1,<2.4' torchvision --index-url https://download.pytorch.org/whl/cpu && \
 pip install -r requirements.txt && \
 python export.py --weights yolov5n.pt --include tflite --int8 --data data/coco128.yaml"
```
If this still produces a bad file, please share the full converter logs and your exact command so we can reproduce; see the TFLite integration guide for notes on quantization specifics. See the Ultralytics TFLite integration docs for details.

**Comment by ozkulah** [comment]
Hi
your first suggestion "using latest docker with tf 2.13" worked but we dont want to get updates directly if there is a change in future. We tried to use previous pinned version "v7.0" but it is too old.  Do you have version number for current latest docker ?

**Comment by pderrenger** [comment]
@ozkulah We don’t publish a numbered tag for “latest”; it’s a rolling build, so pin by digest for reproducibility: 
docker pull ultralytics/yolov5:latest && docker inspect --format='{{index .RepoDigests 0}}' ultralytics/yolov5:latest
Then use ultralytics/yolov5@sha256:<that_digest> in your CI; alternatively build from a specific repo commit with utils/docker/Dockerfile to create your own pinned tag. See the Docker Quickstart wiki for context. ([github.com](https://github.com/ultralytics/yolov5/wiki/Docker-Quickstart))

**Comment by ozkulah** [comment]
Thank you very much , we will do that.

**Comment by pderrenger** [comment]
Glad it helped, @ozkulah—if anything changes with your pinned digest, please reopen and include the digest and full converter logs so we can reproduce.
